presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -444,6 +455,9 @@ protected ObjectListing computeNext(ObjectListing previous)
 
     private Iterator<LocatedFileStatus> statusFromObjects(List<S3ObjectSummary> objects)
     {
+        // NOTE: for encrypted objects, S3ObjectSummary.size() used below is NOT correct,
+        // however, to get the correct size we'd need to make an additional request to get
+        // user metadata, and in this case it doesn't matter.
```

Comment:
Correct — doesn't matter because the encrypted size is always larger, and in this case since it's just used for the split calc.  It matters most when trying to seek within a given file since seeking in S3 must be done relative to the unencrypted size (for instance, to read the last X bytes of a file to pull out metadata the reported size must be the unencrypted size).

Thanks,

-nate

Nate Sammons
Principal Architect
[cid:image002.png@01CFDD78.1CD32F50]

From: David Phillips <notifications@github.com<mailto:notifications@github.com>>
Reply-To: facebook/presto <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, October 27, 2015 at 10:11 AM
To: facebook/presto <presto@noreply.github.com<mailto:presto@noreply.github.com>>
Cc: Nate Sammons <nate.sammons@nasdaq.com<mailto:nate.sammons@nasdaq.com>>
Subject: Re: [presto] Add support for S3 EncryptionMaterialsProvider to PrestoS3FileSystem (#3802)


In presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java<https://github.com/facebook/presto/pull/3802#discussion_r43144183>:

> @@ -444,6 +455,9 @@ protected ObjectListing computeNext(ObjectListing previous)
>
>      private Iterator<LocatedFileStatus> statusFromObjects(List<S3ObjectSummary> objects)
>      {
> +        // NOTE: for encrypted objects, S3ObjectSummary.size() used below is NOT correct,
> +        // however, to get the correct size we'd need to make an additional request to get
> +        // user metadata, and in this case it doesn't matter.


Is this "doesn't matter" because the encrypted size is always larger? The Hive connector computes splits based on this size, so if the encrypted size was smaller, it would be possible to not generate a split for part of the file and thus miss some data.

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/presto/pull/3802/files#r43144183>.
***********************************************************
CONFIDENTIALITY NOTICE: This e-mail and any attachments are for the exclusive and confidential use of the intended recipient and may constitute non-public information. If you received this e-mail in error, disclosing, copying, distributing or taking any action in reliance of this e-mail is strictly prohibited and may be unlawful. Instead, please notify us immediately by return e-mail and promptly delete this message and its attachments from your computer system. We do not waive any work product or other applicable legal privilege(s) by the transmission of this message.
***********************************************************